### PR TITLE
dts: bindings: pwm custom macro for STM32 vendor

### PR DESCRIPTION
**Pull request by FRASTM**
Change the name of the custom macro defined for the stm32 devices to fit the VND_PWM_xxx model
This is **STM32_PWM_COMPLEMENTARY**

Signed-off-by: Francois Ramu <francois.ramu@st.com>

### CODE REVIEW

**Review by GeorgeCGV — DISMISSED** [review]

File: include/zephyr/dt-bindings/pwm/stm32_pwm.h [hunk 1]
```diff
@@ -16,10 +16,10 @@
  * @{
  */
 /** PWM complementary output pin is enabled */
-#define PWM_STM32_COMPLEMENTARY	(1U << 8)
+#define STM32_PWM_COMPLEMENTARY	(1U << 8)
 
 /** @cond INTERNAL_HIDDEN */
-#define PWM_STM32_COMPLEMENTARY_MASK	0x100
+#define STM32_PWM_COMPLEMENTARY_MASK	0x100
```
Comment (erwango) [inline include/zephyr/dt-bindings/pwm/stm32_pwm.h:22]: Let's make change non breaking
```suggestion
#define PWM_STM32_COMPLEMENTARY_MASK	__DEPRECATED_MACRO 0x100
#define STM32_PWM_COMPLEMENTARY_MASK	0x100
```

File: include/zephyr/dt-bindings/pwm/stm32_pwm.h [hunk 2]
```diff
@@ -16,10 +16,10 @@
  * @{
  */
 /** PWM complementary output pin is enabled */
-#define PWM_STM32_COMPLEMENTARY	(1U << 8)
+#define STM32_PWM_COMPLEMENTARY	(1U << 8)
```
Comment (erwango) [inline include/zephyr/dt-bindings/pwm/stm32_pwm.h:19]: Let's make change non breaking
```suggestion
#define PWM_STM32_COMPLEMENTARY	__DEPRECATED_MACRO (1U << 8)
#define STM32_PWM_COMPLEMENTARY	(1U << 8)
```

**Review by erwango — CHANGES_REQUESTED** [review]

File: include/zephyr/dt-bindings/pwm/stm32_pwm.h (hunk 1 above)
Comment (FRASTM) [reply include/zephyr/dt-bindings/pwm/stm32_pwm.h]: ok

File: include/zephyr/dt-bindings/pwm/stm32_pwm.h (hunk 2 above)
Comment (FRASTM) [reply include/zephyr/dt-bindings/pwm/stm32_pwm.h]: ok

**Comment by FRASTM** [comment]
keeping old macro for compatibility

**Review by erwango — DISMISSED** [review]

**Review by GeorgeCGV — DISMISSED** [review]

**Comment by FRASTM** [comment]
fixing compliance error

**Review by GeorgeCGV — DISMISSED** [review]

**Review by erwango — DISMISSED** [review]

File: include/zephyr/dt-bindings/pwm/stm32_pwm.h [hunk 3]
```diff
@@ -16,10 +16,12 @@
  * @{
  */
 /** PWM complementary output pin is enabled */
-#define PWM_STM32_COMPLEMENTARY	(1U << 8)
+#define PWM_STM32_COMPLEMENTARY	(1U << 8) __DEPRECATED_MACRO
```
Comment (henrikbrixandersen) [inline include/zephyr/dt-bindings/pwm/stm32_pwm.h:19]: `__DEPRECATED_MACRO` will not work for definitions used in devicetree.

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
You'll need to fix the `nucleo_l4r5zi` board DTS as well.

File: include/zephyr/dt-bindings/pwm/stm32_pwm.h (hunk 3 above)
Comment (erwango) [reply include/zephyr/dt-bindings/pwm/stm32_pwm.h:19]: Arf, so what are the guidelines to deprecate dts bindings ?

**Comment by FRASTM** [comment]
Changed with @doxygen tag
The PWM_STM32_COMPLEMENTARY_MASK does not need the old style as it is internal
Use the new macro  in the DTS  of the nucleo_l4r5zi board

File: include/zephyr/dt-bindings/pwm/stm32_pwm.h [hunk 4]
```diff
@@ -16,10 +16,14 @@
  * @{
  */
 /** PWM complementary output pin is enabled */
-#define PWM_STM32_COMPLEMENTARY	(1U << 8)
+#define STM32_PWM_COMPLEMENTARY	(1U << 8)
+/**
+ * @deprecated Use the PWM complementary `STM32_PWM_COMPLEMENTARY` flag instead.
+ */
+#define PWM_STM32_COMPLEMENTARY	(1U << 8) __DEPRECATED_MACRO
```
Comment (henrikbrixandersen) [inline include/zephyr/dt-bindings/pwm/stm32_pwm.h:23]: You left the macro here.

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

File: include/zephyr/dt-bindings/pwm/stm32_pwm.h (hunk 4 above)
Comment (FRASTM) [reply include/zephyr/dt-bindings/pwm/stm32_pwm.h]: ... that's like in the commit you mention
capture

Comment (FRASTM) [reply include/zephyr/dt-bindings/pwm/stm32_pwm.h]: removing "__DEPRECATED_MACRO" on PWM_STM32_COMPLEMENTARY

**Review by erwango — APPROVED** [review]
Doc build issue seems unrelated

**Review by henrikbrixandersen — APPROVED** [review]

**Pull request merged**